src/Terrain/MapMaster.java

```diff
@@ -92,15 +99,30 @@ public MapMaster(CommandingOfficers.Commander[] COs, MapInfo mapInfo)
         }
       }
 
-      // Warn if the CO still doesn't have a valid HQ.
-      if( COs[co].HQLocations.isEmpty() )
-      {
-        System.out.println("Warning! CO " + co + " does not have any HQ assigned!");
-        initOK = false;
-        break;
-      }
-    }
-  }
+      if( hasHQ )
+        continue; // If we have an HQ already, we know what our most strategic assets are
+
+      // If we don't have an HQ and do have Labs, then Labs are our HQs
+      if( hasLab )
+        for( XYCoord coord : COs[co].ownedProperties )
+        {
+          if( getEnvironment(coord).terrainType == TerrainType.LAB )
+          {
+            COs[co].HQLocations.add(coord);
+          }
+        }
+      // If we own some property, call that good enough for our purposes
+      else if( hasProperty )
+        COs[co].HQLocations.addAll(COs[co].ownedProperties);
+      // If we don't have property, use our first unit's starting location
+      else if( !COs[co].units.isEmpty() )
+        COs[co].HQLocations.add(new XYCoord(COs[co].units.get(0)));
```

Comment:
Why even bother setting the HQ when there isn't an HQ?

Reply:
The idea in my head was "to provide AIs with something to fall back on re: objectives e.g. if everything is covered by fog".
I don't have super strong opinions on it, but it seems useful as a "avoid sitting around picking your nose" thing.

Reply:
Hm. Seems like an odd edge case. I guess we can just let it ride and see what happens.